### ISSUE TITLE
How do I use visitor in oxc-parser using typescript ?

### ISSUE THREAD

**Issue by rizrmd** [oxc-project/oxc#6244]
Provided example is using rust, how do I do it in typescript/js ?

**Comment by Boshen** [comment]
@rizrmd where have you looked and what is lacking from our current documentation?

**Comment by rizrmd** [comment]
> @rizrmd where have you looked and what is lacking from our current documentation?

Mainly here: https://oxc.rs/docs/guide/usage/parser.html
I did search issues, but most of them are discussing rust based visitor, 

Also, there are prior works from swc like woodpile and swc-walk. 

- Woodpile use wasm-pack to call rust visitor using js. 
- swc-walk wraps acorn-walk using custom visitor that conforms with swc terms.

Both are good enough, the problem the size in wasm.
swc takes +/- 15-20mb uncompressed, while oxc only 600kb. I am impressed!

I tried to parse oxc result using swc-walk, surprisingly it works for simple code. But there are some differences in node.type, and several node properties. I just add/modify them to works, but I think this is still a hack...
